packages/instant/src/components/animations/slide_up_and_down_animation.tsx

```diff
@@ -0,0 +1,100 @@
+import * as React from 'react';
+
+import { keyframes, styled } from '../../style/theme';
+
+const slideKeyframeGenerator = (fromY: string, toY: string) => keyframes`
+    from {
+        position: relative;
+        top: ${fromY}
+    }
+
+    to {
+        position: relative;
+        top: ${toY}
+    }
+`;
+
+export interface SlideAnimationProps {
+    keyframes: string;
+    animationType: string;
+    animationDirection?: string;
+}
+export const SlideAnimation =
+    styled.div <
+    SlideAnimationProps >
+    `
+    animation-name: ${props => props.keyframes};
+    animation-duration: 0.3s;
+    animation-timing-function: ${props => props.animationType};
+    animation-delay: 0s;
+    animation-iteration-count: 1;
+    animation-fill-mode: ${props => props.animationDirection || 'none'};
+    position: relative;
+    z-index: -1;
+`;
+
+export const SlideUpAnimationComponent: React.StatelessComponent<{ downY: string }> = props => (
+    <SlideAnimation animationType="ease-in" keyframes={slideKeyframeGenerator(props.downY, '0px')}>
+        {props.children}
+    </SlideAnimation>
+);
+
+export const SlideDownAnimationComponent: React.StatelessComponent<{ downY: string }> = props => (
+    <SlideAnimation
+        animationDirection="forwards"
+        animationType="cubic-bezier(0.25, 0.1, 0.25, 1)"
+        keyframes={slideKeyframeGenerator('0px', props.downY)}
+    >
+        {props.children}
+    </SlideAnimation>
+);
+
+export interface SlideUpAndDownAnimationProps {
+    delayMs: number;
+    downY: string;
+}
+
+interface SlideUpAndDownState {
+    slideState: 'up' | 'down';
+}
+
+export class SlideUpAndDownAnimationComponent extends React.Component<
+    SlideUpAndDownAnimationProps,
+    SlideUpAndDownState
+> {
+    private _timeoutNumber: number | undefined;
+
+    constructor(props: SlideUpAndDownAnimationProps) {
+        super(props);
+        this._timeoutNumber = undefined;
+        this.state = {
+            slideState: 'up',
+        };
+    }
+
+    public render(): JSX.Element {
```

Comment:
Curious why you decided to return `JSX.Element`. We usually return `React.ReactNode`, but don't know what the functional difference is. Is `JSX.Element` a stricter sub-set of `React.ReactNode`? If so, I like it 😃

Reply:
Don't feel strongly about this, my logic was: when I inspected a rendered StyledComponent using TS, it renders a `JSX.Element`, so that's what I used to determine the type definition:

![image](https://user-images.githubusercontent.com/210914/46968418-00005c80-d068-11e8-97ba-74939a501354.png)

I'm not sure about the functional difference, and a quick pass at googling didn't reveal too much.  Not sure if it's worth more time right now -- in the mean time, I'm down to follow the standard of `React.ReactNode`

Reply:
Thanks for the explanation, let's standardize for now.